src/Microsoft.DotNet.Interactive.Tests/KernelConnectionTestsBase.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using FluentAssertions;
+
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Connection;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+using Microsoft.DotNet.Interactive.Utility;
+using Pocket;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Interactive.Tests;
+
+public abstract class KernelConnectionTestsBase<T>: IDisposable
+{
+    private readonly CompositeDisposable _disposables = new();
+
+    protected KernelConnectionTestsBase(ITestOutputHelper output)
+    {
+        _disposables.Add(output.SubscribeToPocketLogger());
+    }
+
+    protected void RegisterForDisposal(IDisposable disposable)
+    {
+        _disposables.Add(disposable);
+    }
+
+    public void Dispose()
+    {
+        _disposables.Dispose();
+    }
+
+    [WindowsFact]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "Test only enabled on windows platforms")]
+    public async Task it_can_reuse_connection_for_multiple_proxy_kernel()
+    {
+        var configuration = CreateConnectionConfiguration();
+       
+        var connector = await CreateConnectorAsync(configuration);
```

Comment:
Should this not be diposable? I noticed that `NamedPipeKernelConnector` creates a `MultiplexingKernelCommandAndEventReceiver` which is disposable, but never disposes it.

Reply:
I think that kernelConnectors should be all disposable.